SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -1113,6 +1147,199 @@ FString FSpatialGDKEditorToolbarModule::GetOptionalExposedRuntimeIP() const
 	}
 }
 
+FReply FSpatialGDKEditorToolbarModule::OnLaunchDeployment()
+{
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+
+	if (!SpatialGDKSettings->IsDeploymentConfigurationValid())
+	{
+		OnShowFailedNotification(TEXT("Deployment configuration is not valid."));
+
+		return FReply::Handled();
+	}
+
+	AddDeploymentTag(TEXT("dev_login"));
+	
+	if (SpatialGDKSettings->IsGenerateSchemaEnabled())
+	{
+		SpatialGDKEditorInstance->GenerateSchema(false);
+	}
+
+	if (SpatialGDKSettings->IsGenerateSnapshotEnabled())
+	{
+		SpatialGDKGenerateSnapshot(GEditor->GetEditorWorldContext().World(), SpatialGDKSettings->GetSpatialOSSnapshotToSave());
+	}
+
+	TSharedRef<FSpatialGDKPackageAssembly> PackageAssembly = SpatialGDKEditorInstance->GetPackageAssemblyRef();
+	PackageAssembly->OnSuccess.BindRaw(this, &FSpatialGDKEditorToolbarModule::OnBuildSuccess);
+	PackageAssembly->BuildAllAndUpload(
+		SpatialGDKSettings->GetAssemblyName(),
+		SpatialGDKSettings->AssemblyWindowsPlatform,
+		SpatialGDKSettings->AssemblyBuildConfiguration,
+		TEXT(""),
+		SpatialGDKSettings->bForceAssemblyOverwrite
+	);
+
+	return FReply::Handled();
+}
+
+void FSpatialGDKEditorToolbarModule::OnBuildSuccess()
+{
+	const USpatialGDKEditorSettings* SpatialGDKSettings = GetDefault<USpatialGDKEditorSettings>();
+	if (SpatialGDKSettings->IsSimulatedPlayersEnabled())
+	{
+		IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+		FString BuiltWorkerFolder = GetDefault<USpatialGDKEditorSettings>()->GetBuiltWorkerFolder();
+		FString BuiltSimPlayersName = TEXT("UnrealSimulatedPlayer@Linux.zip");
+		FString BuiltSimPlayerPath = FPaths::Combine(BuiltWorkerFolder, BuiltSimPlayersName);
+
+		if (!PlatformFile.FileExists(*BuiltSimPlayerPath))
+		{
+			FString MissingSimPlayerBuildText = FString::Printf(TEXT("Warning: Detected that %s is missing. To launch a successful SimPlayer deployment ensure that SimPlayers is built and uploaded.\n\nWould you still like to continue with the deployment?"), *BuiltSimPlayersName);
+			EAppReturnType::Type UserAnswer = FMessageDialog::Open(EAppMsgType::YesNo, FText::FromString(MissingSimPlayerBuildText));
```

Comment:
Can we use `OnShowFailedNotification` function here?

Reply:
Applied